addons/account/tests/test_account_move_in_invoice.py

```diff
@@ -1551,6 +1551,34 @@ def test_in_invoice_payment_register_wizard(self):
             self.assertFalse(wiz_form._get_modifier('group_payment', 'invisible'))
             self.assertFalse(wiz_form._get_modifier('group_payment', 'readonly'))
 
+        # We can also force the registration of the payment of a draft move with a button hidden
+        # in the gear icon menu.
+        move = self.env['account.move'].create({
+            'move_type': 'in_invoice',
+            'partner_id': self.partner_a.id,
+            'invoice_date': fields.Date.from_string('2023-01-30'),
+            'currency_id': self.other_currency.id,
+            'invoice_payment_term_id': self.pay_terms_b.id,
+            'invoice_line_ids': [
+                Command.create({
+                    'product_id': self.product_line_vals_1['product_id'],
+                    'product_uom_id': self.product_line_vals_1['product_uom_id'],
+                    'price_unit': self.product_line_vals_1['price_unit'],
+                    'tax_ids': [Command.set(self.product_line_vals_1['tax_ids'])],
+                }),
+            ],
+        })
+        action_data = move.action_force_register_payment()  # should not raise an error on non-posted move
+        self.assertTrue(action_data)
+
+        # If the move is already fully paid, we should alert the user
+        move.action_post()
+        reversal = move._reverse_moves()
+        reversal.invoice_date = fields.Date.from_string('2023-01-31')
+        reversal.action_post()
```

Comment:
Why do we need all this in order to test the assertion below?
Shouldn't it fail because we already called `action_force_register_payment`?